pkg/providers/aws/provider_postgres.go

```diff
@@ -61,6 +61,7 @@ const (
 	defaultPostgresUserKey               = "user"
 	defaultPostgresPasswordKey           = "password"
 	defaultStorageEncrypted              = true
+	defalutApplyImmediately              = true
```

Comment:
should it update immediately in all cases or is it something we want to be able to override in certain cases, e.g. RHOAM?

Reply:
its only applied when an engine upgrade is detected

Reply:
yeah i think that's my question. I guess I'm thinking in the case of non rhoam users. that they may want to allow for the engine upgrades to only happen in the maintenance window. so we should have a way to override that that integreatly/rhoam can use.

Reply:
Made this configurable via CR, updated the verification steps to include the CR changes.